### ISSUE TITLE
spanner: allow users to disable internal retrying behaviour for Aborted or SessionNotFound errors

### ISSUE THREAD

**Issue by hengfengli** [googleapis/google-cloud-go#1971]
In Java client library, we have a way to manually manage transactions. 

https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/spanner/DatabaseClient.html#transactionManager--
https://googleapis.dev/java/google-cloud-clients/latest/com/google/cloud/spanner/TransactionManager.html

However, Go client lib does not support it. 

Can we have a similar approach for advanced use cases, e.g., allow users to disable internal retrying behaviour for Aborted or SessionNotFound errors? 

@skuruppu @olavloite Any thoughts?

**Comment by olavloite** [comment]
Technically, it would be possible, and for anyone who would like to create a connection/statement based API on top of the client library, it would make sense. The JDBC driver uses this possibility in the Java client. So for such use cases it does make sense.

It does however also generate questions. Many users are unaware of the possibility that Cloud Spanner might abort any transaction at any time, and that they need to implement their own retry loops and safeguards against this. I find it difficult to think of any valid use case where a normal end user that is just using the client library would need this possibility.

So to sum up my opinion:
1. For anyone who wants to develop a different client library / API on top of the client library, it does make sense.
1. For end users I don't think it adds actual value, as I can't think of anything that they cannot achieve with the current API, unless they belong to category 1 above.

So if we have users in category 1, then we should consider implementing this.

**Comment by hengfengli** [comment]
@olavloite Thanks for your detailed comment. 

@rakyll is developing a `database/sql` driver for spanner: https://github.com/rakyll/go-sql-driver-spanner. The internal retrying behaviour causes a big trouble over there. So it would be nice to have an option to disable the internal retrying to support ORM.
